doc/mgr/orchestrator_cli.rst

```diff
@@ -8,54 +8,239 @@ Orchestrator CLI
+|                                      | “service instance” instead           |
++--------------------------------------+--------------------------------------+
+
 Configuration
 =============
 
-You can select the orchestrator module to use with the ``set backend`` command:
-
-::
+You can select the orchestrator module to use with the ``set backend`` command::
 
     ceph orchestrator set backend <module>
 
-For example, to enable the Rook orchestrator module and use it with the CLI:
-
-::
+For example, to enable the Rook orchestrator module and use it with the CLI::
 
     ceph mgr module enable orchestrator_cli
     ceph mgr module enable rook
     ceph orchestrator set backend rook
 
 
-You can then check backend is properly configured:
-
-::
+You can then check backend is properly configured::
 
     ceph orchestrator status
 
 
 Usage
 =====
 
+.. warning::
+
+    The orchestrator CLI is unfinished and work in progress. Some commands will not
+    exist, or return a different result.
+
+.. note::
+
+    Orchestrator modules may only implement a subset of the commands listed below.
+    Also, the implementation of the commands are orchestrator module dependant and will
+    differ between implementations.
+
+Status
+~~~~~~
+
+::
+
+    ceph orchestrator status
+
+Show current orchestrator mode and high-level status (whether the module able
+to talk to it)
+
+Also show any in-progress actions.
+
+..
+
+    Host Management
+    ~~~~~~~~~~~~~~~
+
+    List hosts accociatted with the cluster: :
+
+        ceph orchestrator host ls
+
+    Add and remove hosts: :
+
+      ceph orchestrator host add <host>
+      ceph orchestrator host rm <host>
+
+    .. note::
+
+    Removing a host only succeeds, if the host is unused.
+
+
+OSD Management
+~~~~~~~~~~~~~~
+
+List Devices
+^^^^^^^^^^^^
+
 Print a list of discovered devices, grouped by node and optionally
 filtered to a particular node:
 
 ::
 
-    orchestrator device ls [node]
+    ceph orchestrator device ls [--host=...] [--devname=...] [--refresh]
 
-Query the status of a particular service (mon, osd, mds, rgw).  For OSDs
-the id is the numeric OSD ID, for MDS services it is the filesystem name:
+Create OSDs
+^^^^^^^^^^^
 
-::
+Create OSDs on a group of devices on a single host::
+
+    ceph orchestrator osd create <host> <ceph-volume-invocation…>
+
+See :ref:`ceph-volume-invocation… <ceph-volume-overview>` for details. E.g.
+``ceph orchestrator osd create host1 lvm create …``
+
+Flags include::
+
+    --replace-osd-ids=previous_osd_ids  # comma separated list. See notes below
+    --objectstore={filestore,bluestore}
+    --dmcrypt
+    --osds-per-device <N>
+    --shared-device <dev>  # for hybrid osds, db or journal
+    --num-slots <N>   # if we want to divide shared device by N > num devices
+    --shared-underprovision-ratio <ratio>
+
+
+.. option:: --replace-osd-ids
 
-    orchestrator service status <type> <id>
+    When ``--replace-osd-ids`` is used, behavior is identical to ``osd
```

Comment:
Yes, `batch` doesn't have anything like this, and it would involve a non-trivial amount of work to try it out.

Reply:
adding it to the todo list at https://pad.ceph.com/p/ceph-dashboard&orchestrator-f2f-2018-11 .. @sebastian-philipp where is the todo list being maintained?  on the trello board?

Reply:
@liewegas I've updated the Trello Board https://trello.com/b/hzk7dotY/ceph-orchestrator

Reply:
I think that proposed items for ceph-volume should have some sort of elaboration, otherwise we keep having these disconnects. Happy to meet and discuss these flags and ideas further.